pkg/chunk/elastic/es_index_client.go

```diff
@@ -0,0 +1,344 @@
+package elastic
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"flag"
+	"github.com/cortexproject/cortex/pkg/chunk"
+	chunk_util "github.com/cortexproject/cortex/pkg/chunk/util"
+	"github.com/cortexproject/cortex/pkg/util"
+	"github.com/go-kit/kit/log/level"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"reflect"
+
+	"fmt"
+	"github.com/olivere/elastic"
+)
+
+const (
+	null         = string('\xff')
+	maxFetchDocs = 1000
+)
+
+// Config for a BoltDB index client.
+type Config struct {
+	Address       string `yaml:"address"`
+	IndexType     string `yaml:"index_type"`
+	User          string `yaml:"user"`
+	Password      string `yaml:"password"`
+	TLSSkipVerify bool   `yaml:"tls_skip_verify"`
+	CertFile      string `yaml:"cert_file"`
+	KeyFile       string `yaml:"key_file"`
+	CaFile        string `yaml:"ca_file"`
+}
+
+// RegisterFlags registers flags.
+func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
+	f.StringVar(&cfg.Address, "elastic.address", "http://127.0.0.1:9200", "Address of ElasticSearch.")
+	f.StringVar(&cfg.IndexType, "elastic.index_type", "lokiindex", "Index Type used in ElasticSearch.")
+	f.StringVar(&cfg.User, "elastic.user", "user", "User used in ElasticSearch Basic Auth.")
+	f.StringVar(&cfg.Password, "elastic.password", "password", "Password used in ElasticSearch Basic Auth.")
+	f.BoolVar(&cfg.TLSSkipVerify, "elastic.tls_skip_verify", true, "Skip tls verify or not. Default is skip.")
```

Comment:
in my opinion the default value for tls skip verify should be false. False is the securest option and if someone really wants to use something else, he knows what he does.

Reply:
In my opinion, most of the customers are not care about tls verify and what they want is to use https the same as http, which means they need not do anything.

Reply:
It is a better security practice to have this as false by default.